mitmproxy/net/tls.py

```diff
@@ -106,6 +107,13 @@ def _create_ssl_context(
     # Options
     context.set_options(DEFAULT_OPTIONS)
 
+    # ECDHE for Key exchange
+    if ctx.options.curve is not None:
```

Comment:
The return values of methods in `net/tls.py` should depend on their arguments only, not on externally-defined variables/options. This needs to be accessed in tlsconfig.py and passed in as an argument.